march_hardware/src/Joint.cpp

```diff
@@ -133,7 +133,7 @@ double Joint::getPosition() const
 
 double Joint::getVelocity() const
 {
-  return this->velocity_;
+  return this->imc_->getVelocityRadIncremental();
```

Comment:
I think it is better to return this->velocity_ here and calculate the velocity in Joint::readEncoders. Also use whichever encoder has a higher resolution. That way, holes in encoder data are patched with this velocity instead of the velocity as we calculated it previously (see line 120-126)

Reply:
Yes, I just did.